lib/truffle/java/core_ext/module.rb

```diff
@@ -44,15 +44,15 @@ def include_package(package)
     @included_packages = [ package ]
     @java_aliases ||= {}
 
-    def self.const_missing(constant)
+    define_singleton_method :const_missing do |constant|
```

Comment:
This is slower and inherit a scope it should not, I am against it.
The cop must be broken, only `def meth` is problematic, not `def singleton.meth`.